reports/generic-query-based-data-viewer.md

```diff
@@ -0,0 +1,154 @@
+has the following features:
+
+- Before using the app, a developer defines via a config file the SPARQL queries that should be used by the app.
+- Users can log in with their WebID or identity provider.
+- Users select the query based on their name from the sidebar.
+- The app shows the results of the query in a table. 
+- Specific query variables influence how the app shows the results:
+  - If a variable is a link to an image and the variable ends with `_img` then the app shows the actual images.
+  - If a variable is a float and the variable ends with `_float`, the app shows the values as floats.
+- Users can refresh the results of a query via a button.
+
+You find a screencast [here](https://cloud.ilabt.imec.be/index.php/s/AJomCGpLjYbxmCX).
+
+<!--
+Provide a list of important technical decisions and assumptions.
+-->
+We made the following important technological decisions and assumptions:
+
+- The app has the [option](https://github.com/SolidLabResearch/generic-data-viewer/tree/ccccc3aa882bb7c1c76fdafdf5e937ffc491d8af#configuration-file) 
+  to redirect requests from Comunica, which executes the queries, via a proxy. 
+  The app does this to deal with data sources that don't set the CORS headers correctly.
+
+## User flow
+
+<!--
+Describe a concrete user flow with the approved solution.
+Complete the following sections:
+-->
+
+### Actors/actresses
+
+- User of the application.
+
+### Preconditions
+
+- The user has Node.js installed.
+
+### Steps
+
+1. Clone the repository (v1.0.0) via
+   ```shell
+   git clone -b v1.0.0 https://github.com/SolidLabResearch/generic-data-viewer.git
+   ```
+2. Install the dependencies via
+   ```shell
+   npm i 
+   ```
+3. Prepare and start the Community Solid Server with the pods via
+   ```shell
+   npm run prepare:pods && npm run start:pods
+   ```
+   The server is ready when the following message appears:
+   ```
+   Listening to server at http://localhost:3000/
+   ```
+4. Open another terminal.
+5. Start the app via
+   ```shell
+   npm start
+   ```
+   The app is ready when the following message appears:
+   ```
+   Running CORS Anywhere on 0.0.0.0:8000
+   ```
+6. Open another terminal.
+7. Start the proxy via
+   ```shell
+   npm run start:proxy
+   ```
+   The app is ready when the following message appears:
+   ```
+   Running CORS Anywhere on 0.0.0.0:8000
+   ```
+8. Navigate to <http://localhost:3000>.
+9. Click on "Test query" in the sidebar.
+10. The results appear in the table on the right.
+11. Click on "A list of my favourite books".
+12. The app shows the message "Something went wrong while preparing the query."
+    This happens because the data source of this query requires you to be authenticated.
+13. Log in with <http://localhost:8080> as identity provider.
+14. Click on "A list of my favourite books".
+15. The results appear in the table on the right.
+
+### Postconditions
+
+None.
+
+## Follow-up actions
```

Comment:
(linked to my comment [above](https://github.com/SolidLabResearch/Challenges/pull/122/files#r1296752093) ): maybe a general 'what can I as a reader do with this result', i.e. some call to action as how to adapt this result to your use case

Reply:
Isn't that clear from the problem section? If not then there might be an issue with the problem section.

Reply:
But how can we do this without knowing your exact use case (and the other people's use cases)?

Reply:
I'd just link to the (part of) the solution code readme that shows how to configure/adapt it

Reply:
Fixed [here](https://github.com/SolidLabResearch/Challenges/pull/122/commits/f3a774645a253a387a558b5300c59c3a01df87e2#diff-789ddf355d65b22d46d58b108db7d4d7696fcaccf830e341b1ba873b3c972e00R50-R52).